src/webgpu/api/validation/encoding/createRenderBundleEncoder.spec.ts

```diff
@@ -0,0 +1,177 @@
+export const description = `
+createRenderBundleEncoder validation tests.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import {
+  kColorTextureFormats,
+  kDepthStencilFormats,
+  kTextureFormatInfo,
+} from '../../../capability_info.js';
+import { ValidationTest } from '../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('attachment_formats')
+  .desc(
+    `
+    Tests that createRenderBundleEncoder correctly validates the attachment color formats passed to it.
+      - Must accept any renderable color formats in colorFormats
+      - Must accept any depth stencil format in depthStencilFormat
+      - Must be <= 8 colorFormats
+    `
+  )
+  .params(u =>
+    u //
+      .combine('colorFormat', kColorTextureFormats)
```

Comment:
Spec says it needs to be both Color AND renderable.

Maybe should test all formats and merge into the invalid_color_formats test. So we get coverage of both depth stencil and non-renderable color.

Should be an OK number of paramterizations if we test all formats for color, and all formats for depth. But NOT the cross product of all pairings for color/depth formats.

Reply:
This test does set the pass/fail expectation on whether the format is renderable or not.

> But NOT the cross product of all pairings for color/depth formats.

Not sure that I follow?

Reply:
sorry, to be more clear

we can test:

```typescript
.combine('colorFormat', kAllTextureFormats)
.fn(t => {
   const bool success = isColorRenderable(t.params.colorFormat); // made-up function.

});
```
and
```typescript
.combine('depthStencilFormat', kAllTextureFormats)
.fn(t => {
   const bool success = isDepthStencil(t.params.depthStencilFormat); // made-up function.

});
```

but the would be unnecessary and way too many parameterizations 
```typescript
.combine('colorFormat', kAllTextureFormats)
.combine('depthStencilFormat', kAllTextureFormats)
.fn(t => {
   const bool success = isColorRendeable(t.params.colorFormat) && isDepthStencil(t.params.depthStencilFormat); // made-up function.

});
```

Reply:
Took a crack at it. Now all the format testing happens in one fairly tight test and the attachment state testing (colorFormats count, depthStencilFormat requirements) happens in another that doesn't iterate through all the format combinations. Executes a lot faster now. 😆